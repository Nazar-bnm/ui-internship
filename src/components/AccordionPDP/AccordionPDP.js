import React from 'react';

import Accordion from '^/Accordion';
import ContactDetails from '^/Footer/ContactDetails';
import { accordionItemsData } from './dataItems';

import './AccordionPDP.scss';

const CN = 'AccordionPDP';

const AccordionPDP = () => (
  <div className={CN}>
    <Accordion data={accordionItemsData} heightItem="100px" />
    <ContactDetails title="share" />
  </div>
);

export default AccordionPDP;
