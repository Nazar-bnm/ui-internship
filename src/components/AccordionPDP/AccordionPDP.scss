@import '../../scss/global_styles';

.AccordionPDP {
  width: 300px;

  .contact-block {
    border-top: 1px solid $silver-chalice;

    &__title {
      color: $black;
    }

    &__item {
      transition: background-color linear 0.5s;
      padding: 3px;
      background-color: $black;
      color: $white;

      &:hover,
      &:active {
        background-color: $silver-chalice;
      }

      a {
        text-decoration: none;
      }
    }

    i.icon {
      font-size: 1rem;
      color: $white;
    }
  }

  .accordion-item__description {
    font-size: 0.9rem;
  }
}
