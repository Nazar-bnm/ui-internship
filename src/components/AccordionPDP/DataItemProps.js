export const accordionItemsData = [
  {
    id: 1,
    title: 'description',
    description: ` Architecto, earum temporibus quidem ex eaque unde harum sit, 
      deleniti tempore laboriosam neque cumque!Architecto, earum temporibus 
      quidem ex eaque unde harum sit, deleniti tempore laboriosam neque cumque!`,
  },
  {
    id: 2,
    title: 'additional info',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit. 
      `,
  },
  {
    id: 3,
    title: 'tags',
    description:
      `Earum temporibus quidem ex eaque unde harum sit, 
      `,
  },
];
