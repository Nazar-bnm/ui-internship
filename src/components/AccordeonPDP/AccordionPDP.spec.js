import React from 'react';
import { shallow } from 'enzyme';
import AccordionPDP from './AccordionPDP';

describe('<AccordionPDP />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<AccordionPDP />);
  });

  it('should match snapshot <AccordionPDP />', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
