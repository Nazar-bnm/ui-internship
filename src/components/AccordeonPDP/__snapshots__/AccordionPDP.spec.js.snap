// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccordionPDP /> should match snapshot <AccordionPDP /> 1`] = `
<div
  className="AccordionPDP"
>
  <Accordion
    className=""
    data={
      Array [
        Object {
          "description": " Architecto, earum temporibus quidem ex eaque unde harum sit, 
      deleniti tempore laboriosam neque cumque!Architecto, earum temporibus 
      quidem ex eaque unde harum sit, deleniti tempore laboriosam neque cumque!",
          "id": 1,
          "title": "description",
        },
        Object {
          "description": "Lorem ipsum dolor sit amet consectetur adipisicing elit. 
      ",
          "id": 2,
          "title": "additional info",
        },
        Object {
          "description": "Earum temporibus quidem ex eaque unde harum sit, 
      ",
          "id": 3,
          "title": "tags",
        },
      ]
    }
    heightItem="100px"
    open={false}
    scroll={false}
    showAll={true}
  />
  <ContactDetails
    title="share"
  />
</div>
`;
